luigi/target.py

```diff
@@ -348,3 +349,20 @@ def __exit__(self, exc_type, exc, traceback):
         if exc_type:
             return
         return super(AtomicLocalFile, self).__exit__(exc_type, exc, traceback)
+
+
+class LazyFormatMixin:
+    @property
+    def format(self):
+        if self._format is None:
+            self._set_default_format()
+        return self._format
+
+    def _set_default_format(self, mode=None):
```

Comment:
It's odd to me that the method `_set_default_format` take an optional arg of `mode`...